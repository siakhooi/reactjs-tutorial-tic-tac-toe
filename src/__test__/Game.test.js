import React from 'react'
import renderer from 'react-test-renderer'
import Game from '../Game'

const ticXY = (t, x, y) => {
  t.children[0] // <div className="game-board">
    .children[0]
    .children[x] // div
    .children[y] // <div className="board-row">
    .props.onClick()
}
const jumpX = (t, x) => {
  t.children[1] // <div className="game-info">
    .children[1] // ol
    .children[x]// li
    .children[0].props.onClick() // button
}
it('Game', () => {
  const component =
    renderer.create(
      <Game />
    )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  ticXY(tree, 0, 0)
  ticXY(tree, 1, 0)
  ticXY(tree, 0, 1)
  ticXY(tree, 1, 1)
  ticXY(tree, 0, 2) // X win
  ticXY(tree, 1, 2) // O still click

  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  jumpX(tree, 2)
  jumpX(tree, 1)

  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
