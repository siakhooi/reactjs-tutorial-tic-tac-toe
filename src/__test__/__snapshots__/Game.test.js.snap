// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game 1`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Next player: X
    </div>
    <ol>
      <li>
        <button
          onClick={[Function]}
        >
          Go to game start
        </button>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Game 2`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          O
        </button>
        <button
          className="square"
          onClick={[Function]}
        >
          O
        </button>
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Winner: X
    </div>
    <ol>
      <li>
        <button
          onClick={[Function]}
        >
          Go to game start
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #1
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #2
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #3
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #4
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #5
        </button>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Game 3`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Next player: O
    </div>
    <ol>
      <li>
        <button
          onClick={[Function]}
        >
          Go to game start
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #1
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #2
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #3
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #4
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          Go to move #5
        </button>
      </li>
    </ol>
  </div>
</div>
`;
